homeassistant/components/climate/homematicip_cloud.py

```diff
@@ -0,0 +1,125 @@
+https://home-assistant.io/components/climate.homematicip_cloud/
+"""
+
+import logging
+
+from homeassistant.components.climate import (
+    ClimateDevice, SUPPORT_TARGET_TEMPERATURE, SUPPORT_OPERATION_MODE)
+from homeassistant.const import TEMP_CELSIUS, ATTR_TEMPERATURE
+from homeassistant.components.homematicip_cloud import (
+    HomematicipGenericDevice, DOMAIN as HOMEMATICIP_CLOUD_DOMAIN,
+    ATTR_HOME_ID)
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_BOOST = 'Boost'
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the HomematicIP climate devices."""
+    from homematicip.group import HeatingGroup
+
+    if discovery_info is None:
+        return
+    home = hass.data[HOMEMATICIP_CLOUD_DOMAIN][discovery_info[ATTR_HOME_ID]]
+
+    devices = []
+    for device in home.groups:
+        if isinstance(device, HeatingGroup):
+            devices.append(HomematicipHeatingGroup(hass, home, device))
+
+    if devices:
+        async_add_devices(devices)
+
+
+class HomematicipHeatingGroup(HomematicipGenericDevice, ClimateDevice):
+    """Representation of a MomematicIP heating group."""
+
+    def __init__(self, hass, home, device):
+        """Initialize heating group."""
+        device.modelType = 'Group'
+        super().__init__(home, device)
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return 'mdi:nest-thermostat'
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._device.setPointTemperature
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._device.actualTemperature
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._device.humidity
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._device.minTemperature
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._device.maxTemperature
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        await self._device.set_point_temperature(temperature)
+
+    @property
+    def current_operation(self):
+        """Return current operation (auto, manual, boost, vacation)."""
+        if self._device.boostMode:
+            return STATE_BOOST
```

Comment:
Only states present in the climate base component are allowed. If you want to add a new state, it needs to pass consensus in an issue in the architecture repo first.

Reply:
@MartinHjelmare STATE_BOOST is also used in homematic climate platform. This is quite useful  for users as this turns the heating on for a few minutes and changes then back to automatic mode. The feature is named in all Homematic(IP) boost mode.

Reply:
That doesn't change how we should do things. Changes to entity model needs to go through review in the architecture repo:
https://developers.home-assistant.io/docs/en/entity_index.html#changing-the-entity-model